### PR TITLE
HTBHF-2303 Replaced the single message schedule with one for each message type

### PR DESCRIPTION
**Pull request by YetAnotherMatt**
Currently we have a single message processor & schedule, so that all messages are processed one after another. With very few messages this is efficient and quick, but with more messages this becomes inefficient, and does not allow parallel processing on multiple instances. It also means that we cannot configure certain message types (such as sending email) to run only at specific times of the day.

Replace the single message processor schedule with one for each message type.

### CODE REVIEW

**Comment by mikeSmith08** [comment]
You might want to move the story for this from Open into In Development. :-)

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    @SchedulerLock(
+            name = "Process ADDITIONAL_PREGNANCY_PAYMENT messages",
+            lockAtLeastForString = FIFTEEN_SECONDS,
+            lockAtMostForString = ONE_HOUR)
+    @NewRequestContextWithSessionId(sessionId = "MessageProcessor:ADDITIONAL_PREGNANCY_PAYMENT")
+    public void processAdditionalPregnancyPaymentMessages() {
+        messageProcessor.processMessagesOfType(MessageType.ADDITIONAL_PREGNANCY_PAYMENT);
+    }
+
+    @Scheduled(cron = "*/30 * 8-18 * * *")  // every 30 seconds between 8am and 6pm
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java]: I'm surprised the schedule isn't in config, is that going to be a subsequent PR?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import uk.gov.dhsc.htbhf.claimant.message.MessageProcessor;
+import uk.gov.dhsc.htbhf.claimant.message.MessageType;
+import uk.gov.dhsc.htbhf.requestcontext.aop.NewRequestContextWithSessionId;
+
+/**
+ * Component responsible for scheduling the processing of messages.
+ * Each MessageType has its own cron schedule - see https://stackoverflow.com/a/26147143 for examples.
+ * Each messageType also has its own minimum and maximum lock period. For a definition of the ISO-8601 duration format, see
+ * https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm.
+ * Note that the {@link MessageProcessor} also specifies a limit on the number of messages to process
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java]: We might also want to think about making this configurable on a message type basis too.

Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java]: True

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java]: I wasn't sure if it needed to be in config. The question is, are we ever going to change this on a per-environment basis? If not, I'm not sure it needs to be in config.

Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java]: Yep, that's fine, we can change when we need to.

**Pull request merged**